src/coreclr/jit/morph.cpp

```diff
@@ -335,14 +335,20 @@ GenTree* Compiler::fgMorphExpandCast(GenTreeCast* tree)
             && tree->gtOverflow()
 #elif defined(TARGET_AMD64)
             // Amd64: src = float, dst = uint64 or overflow conversion.
-            // This goes through helper and hence src needs to be converted to double.
-            && (tree->gtOverflow() || (dstType == TYP_ULONG))
+            // src needs to be converted to double except for the following cases
+            //       dstType = int/uint/ulong for AVX512F
+            //       dstType = int for SSE41
+            // For PRE SSE41, the all src is converted to TYP_DOUBLE
```

Comment:
```suggestion
            // For pre-SSE41, the all src is converted to TYP_DOUBLE
```
Nit: PRE is not an abbreviation, so it should not be all CAPS.